main.cpp

```diff
@@ -1,57 +1,56 @@
 #include <iostream>
 #include "BaseUnit.h"
 
+#include <fstream>
+
 //Everything happens here for now
 int main(int argc, char *argv[])
 {
-    const std::string errormsg = "Incorrect starting arguments.\nYou should start like this.:\nPROGRAM.EXE CHARACTER1 HP DMG CHARACTER2 HP DMG";
-	if (argc != 7){
-		std::cout << errormsg << std::endl;
-		std::cin.get();
-		return 1;
+	if (argc != 3){
+		std::cout << "Incorrect starting arguments.\nYou should start like this:\nPROGRAM.EXE UNITFILE1 UNITFILE2" << std::endl;
+		return 1; //Error in cmdline args
 	}
-	else
+    
+    try
     {
-        try
-        {
-			BaseUnit Unit1(argv[1], std::stoi(argv[2]), std::stoi(argv[3]));
-			BaseUnit Unit2(argv[4], std::stoi(argv[5]), std::stoi(argv[6]));
-
-			BaseUnit* attacker = &Unit1;
-			BaseUnit* defender = &Unit2;
 
-			//Fighting loop, might end up in a separate class later
-			while (true)
-			{
-				std::cout << Unit1.showStats() << std::endl;
-				std::cout << Unit2.showStats();
+		BaseUnit Unit1 = BaseUnit::parseUnit(argv[1]);
+		BaseUnit Unit2 = BaseUnit::parseUnit(argv[2]);
 
-				std::cin.clear();
-				std::cin.get();
+		BaseUnit* attacker = &Unit1;
+		BaseUnit* defender = &Unit2;
 
-				std::cout << attacker->getName() << " -> " << defender->getName() << std::endl;
-				defender->gotHit(*attacker);
-				if (defender->isDead())
-				{
-					break;
-				}
+		//Fighting loop, might end up in a separate class later
+		while (true)
+		{
+			defender->gotHit(*attacker);
+			if (defender->isDead())
+			{
+				break;
+			}
 				
 
-				BaseUnit* tmp = attacker;
-				attacker = defender;
-				defender = tmp;
-			}
-			std::cout << Unit1.showStats() << std::endl;
-			std::cout << Unit2.showStats() << std::endl;
-			std::cout << defender->getName() << " died. " << attacker->getName() << " wins.";
-        }
-        catch (const std::exception& )
-        {
-            std::cout << errormsg << std::endl;
-            std::cin.get();
-            return 1;
-        }
+			BaseUnit* tmp = attacker;
+			attacker = defender;
+			defender = tmp;
+		}
+		std::cout << attacker->getName() << " wins. Remaining HP:" << attacker->getHP() << std::endl;
     }
-
-	std::cin.get();
+	catch (const std::string& file_name )
+	{
+		std::cout << "Unable to open file " << file_name<< std::endl;
+        return 2; //No such file
+    }
+	catch (const std::exception& e)
+	{
+		std::cout << "Invalid file syntax" << std::endl;
+		return 3; //stoi exception
```

Comment:
Ebből a típusból nem derül ki, hogy ez a hiba. Ezért kell precíz exception típusokat használni, és/vagy a hibaüzenetet a dobás helyén beleírni az exceptionbe. Feljebb, ahol elkapódik, a hívó talán nem is tudja, hogy mi a meghívott függvény implementációja, és ott mi dobhat ilyen kivételt. Főleg, hogy minden standard exception az `exception`ből származik, így a típus itt semmit nem árul el.